test/specs/lib/factories-test.tsx

```diff
@@ -530,6 +534,46 @@ describe('factories', () => {
       itOverridesDefaultPropsWithFalseyProps('element', {
         value: <div {...{ undef: undefined, nil: null, zero: 0, empty: '' } as any} />,
       })
+
+      test('applies defaultProps, element props and overrideProps if element type is the same as the functional component', () => {
+        const MyComponent = createMockComponent('MyComponent')
+        testCreateShorthand(
+          {
+            Component: MyComponent,
+            value: MyComponent({ common: 'user', user: true }),
+            defaultProps: { common: 'default', default: true },
+            overrideProps: { common: 'override', override: true },
+          },
+          { common: 'override', user: true, default: true, override: true },
+        )
+      })
+
+      test('applies only the element props and forwards the element as is if element type is different than the primitive component', () => {
```

Comment:
N.A. anymore since the implementation was drastically simplified following decision:
https://github.com/stardust-ui/react/pull/496#discussion_r238708722